ENIGMAsystem/SHELL/Universal_System/Extensions/VideoPlayer/videoplayer.cpp

```diff
@@ -0,0 +1,155 @@
+/*
+
+ MIT License
+ 
+ Copyright © 2020 Samuel Venable
+ 
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated documentation files (the "Software"), to deal
+ in the Software without restriction, including without limitation the rights
+ to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ copies of the Software, and to permit persons to whom the Software is
+ furnished to do so, subject to the following conditions:
+ 
+ The above copyright notice and this permission notice shall be included in all
+ copies or substantial portions of the Software.
+ 
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ SOFTWARE.
+ 
+*/
+
+#include <string>
+#include <thread>
+#include <map>
+
+#include "videoplayer.h"
+
+#include <mpv/client.h>
+
+using std::string;
+
+static std::map<string, string> vidmap;
+static std::map<string, string> widmap;
+
+static std::map<string, string> plymap;
+static std::map<string, string> gfxmap;
+
+static std::map<string, bool>   fllmap;
+static std::map<string, int>    volmap;
+
+void video_loop(mpv_handle *mpv) {
+  while (true) {
+    mpv_event *event = mpv_wait_event(mpv, -1);
+    if (event->event_id == MPV_EVENT_END_FILE) break;
+    if (event->event_id == MPV_EVENT_SHUTDOWN) break;
+  }
+  plymap[std::to_string((unsigned long long)mpv)] = "no";
+  mpv_terminate_destroy(mpv);
+}
+
+namespace enigma_user {
+
+string video_add(string fname) {
+  mpv_handle *mpv = mpv_create();
+  string ind = std::to_string((unsigned long long)mpv);
+  vidmap.insert(std::make_pair(ind, fname)); 
+  return ind;
+}
+
+void video_set_option_string(string ind, string option, string value) {
+  mpv_handle *mpv = (mpv_handle *)stoull(ind, nullptr, 10);
+  mpv_set_option_string(mpv, option.c_str(), value.c_str());
+}
+
+void video_play(string ind) {
+  mpv_handle *mpv = (mpv_handle *)stoull(ind, nullptr, 10);
+  plymap.erase(ind);
+  plymap.insert(std::make_pair(ind, "yes"));
+  mpv_initialize(mpv);
+  const char *cmd[] = { "loadfile", vidmap.find(ind)->second.c_str(), NULL };
+  mpv_command(mpv, cmd);
+  std::thread stringhread(video_loop, mpv);
+  stringhread.detach();
+}
+
+bool video_is_playing(string ind) {
+  return (plymap[ind] == "yes");
```

Comment:
At the point where you're comparing a string to "yes"/"no", you should be storing a boolean and replacing it with (bool_value ? "yes" : "no") in the places where a string is required